FueledUtils/GradientView.swift

```diff
@@ -0,0 +1,250 @@
+///
+public final class GradientView: UIView {
+	public enum Definition {
+		///
+		/// Defines a simple gradient, that has a start color and an end color
+		///
+		case simple(startColor: UIColor, endColor: UIColor)
+		///
+		/// Defines a custom gradient, that has as many colors and location point as wanted.
+		/// It should at least contain 2 elements.
+		/// When used in `GradientView`, it ensures that it is the case, and will report a
+		/// crash at runtime if the array has less than 2 elements.
+		///
+		case custom([(color: UIColor, location: CGFloat)])
+
+		///
+		/// Get the start color of the receiver.
+		/// If the current value is `.simple`, returns `startColor`.
+		/// If the current value is `.custom`, returns the first color of the array. If the array is empty, this will crash at runtime.
+		///
+		public var startColor: UIColor {
+			switch self {
+			case .simple(let startColor, _):
+				return startColor
+			case .custom(let colors):
+				return colors.first!.color
+			}
+		}
+
+		///
+		/// Get the end color of the receiver.
+		/// If the current value is `.simple`, returns `endColor`.
+		/// If the current value is `.custom`, returns the last color of the array. If the array is empty, this will crash at runtime.
+		///
+		public var endColor: UIColor {
+			switch self {
+			case .simple(_, let endColor):
+				return endColor
+			case .custom(let colors):
+				return colors.last!.color
+			}
+		}
+
+		fileprivate var gradientInfo: [(color: UIColor, location: CGFloat)] {
+			switch self {
+			case .simple(let startColor, let endColor):
+				return [
+					(startColor, 0.0),
+					(endColor, 1.0),
+				]
+			case .custom(let colors):
+				return colors
+			}
+		}
+	}
+
+	///
+	/// Defines the different type of supported gradients.
+	///
+	public enum GradientType {
+		///
+		/// Defines a linear gradient type, along the specified direction.
+		/// When used with `GradientView`, `direction` is scaled to the bounds of the view.
+		///
+		/// ## Examples
+		/// - `.linear(direction: CGPoint(x: 0.0, y: 1.0)`: Define a gradient going from the left edge to the right edge.
+		/// - `.linear(direction: CGPoint(x: 1.0, y: 1.0)`: Define a gradient going from the top-left corner to the bottom-right corner.
+		/// - `.linear(direction: CGPoint(x: -1.0, y: -1.0)`: Define a gradient going from the bottom-right corner to the top-left corner.
+		/// - `.linear(direction: CGPoint(x: -1.0, y: 0.0)`: Define a gradient going from the bottom edge to the top edge.
+		/// - `.linear(direction: CGPoint(x: 0.0, y: 0.5)`: Define a gradient going from the top edge to the middle, extending the last color until the bottom edge.
+		///
+		/// - Note: If `direction` is `.zero`, only the `endColor` of the `Definition` will be drawn. See `GradientView.Definition.endColor` for more info
+		/// about what this property refers to.
+		///
+		case linear(direction: CGPoint)
```

Comment:
@stephanecopin I've just realised that for Linear gradient, there should be 2 points generally speaking defining the axis along which the gradient is drawn. Why don't we allow that to happen? We could still encapsulate the notion of direction in simple-to-use constants but my concern is that we make some complex things impossible here and what we want is simple things simple, complex things possible. :)